crates/ruff_dev/src/generate_json_schema.rs

```diff
@@ -20,7 +21,17 @@ pub(crate) struct Args {
 }
 
 pub(crate) fn main(args: &Args) -> Result<()> {
-    let schema = schema_for!(Options);
+    let mut schema = schema_for!(Options);
+    // Remove RUF014 since it's not present without the off-by-default feature `unreachable-code`
+    if let Some(Schema::Object(schema_object)) = &mut schema.definitions.get_mut("RuleSelector") {
```

Comment:
Can we do this in `rule_selector.rs` instead? I think that's where we did this in the past, for similarly gated rules:

```diff
diff --git a/crates/ruff/src/rule_selector.rs b/crates/ruff/src/rule_selector.rs
index 7c8e7903c..f6ea3de4e 100644
--- a/crates/ruff/src/rule_selector.rs
+++ b/crates/ruff/src/rule_selector.rs
@@ -196,15 +196,16 @@ pub(crate) const fn prefix_to_selector(prefix: RuleCodePrefix) -> RuleSelector {

 #[cfg(feature = "schemars")]
 mod schema {
-    use crate::registry::RuleNamespace;
-    use crate::rule_selector::{Linter, Rule, RuleCodePrefix};
-    use crate::RuleSelector;
     use itertools::Itertools;
+    use schemars::JsonSchema;
     use schemars::_serde_json::Value;
     use schemars::schema::{InstanceType, Schema, SchemaObject};
-    use schemars::JsonSchema;
     use strum::IntoEnumIterator;

+    use crate::registry::RuleNamespace;
+    use crate::rule_selector::{Linter, RuleCodePrefix};
+    use crate::RuleSelector;
+
     impl JsonSchema for RuleSelector {
         fn schema_name() -> String {
             "RuleSelector".to_string()
@@ -228,20 +229,6 @@ mod schema {
                     .into_iter()
                     .chain(
                         RuleCodePrefix::iter()
-                            .filter(|p| {
-                                // Once logical lines are active by default, please remove this.
-                                // This is here because generate-all output otherwise depends on
-                                // the feature sets which makes the test running with
-                                // `--all-features` fail
-                                !Rule::from_code(&format!(
-                                    "{}{}",
-                                    p.linter().common_prefix(),
-                                    p.short_code()
-                                ))
-                                .unwrap()
-                                .lint_source()
-                                .is_logical_lines()
-                            })
                             .map(|p| {
                                 let prefix = p.linter().common_prefix();
                                 let code = p.short_code();
```